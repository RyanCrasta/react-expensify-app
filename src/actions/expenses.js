import { v4 as uuidv4 } from 'uuid'; // to generate some random ID

// action generator
// ADD_EXPENSE
const addExpense = ({description='', note='', amount=0, createdAt=0} = {}) => {
    return{
        type: 'ADD_EXPENSE',
        expense:{
            id: uuidv4(),
            description,
            note,
            amount,
            createdAt
        }
    }
}
  
//REMOVE_EXPENSE
const removeExpense = ({id}) => {
    return{
        type: 'REMOVE_EXPENSE',
        id
    }
}

//EDIT_EXPENSE
const editExpense = (id, updates) => {
    return{
        type: 'EDIT_EXPENSE',
        id,
        updates
    }
}

export {addExpense, removeExpense, editExpense}