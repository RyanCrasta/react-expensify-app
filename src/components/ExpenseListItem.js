import React from 'react';

import {Link} from 'react-router-dom';

 
const ExpenseListItem = (props) => {
    return(
        <div>
        {props.expenses.map((exp) => {
            return(
                <div key={exp.id}>
                    <Link to={`/edit${exp.id}`}><h3>{exp.description}</h3></Link>
                    
                    {exp.amount} - {exp.createdAt}
                    
                    <p></p>
       
                </div>
            )

        })}
        </div>
    )
}



export { ExpenseListItem as default }