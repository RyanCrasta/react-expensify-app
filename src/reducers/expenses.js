const expensesReducerDefaultState = [];

// expenses reducer
const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch(action.type){
        case 'ADD_EXPENSE':
            return [...state, action.expense]
            // return state.concat(action.expense); 
            // v dont want ot chnage state but just read state so use concat instead of push
        
        case 'REMOVE_EXPENSE':
            return state.filter((ele) => {
                return(ele.id !== action.id)
            })
        
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if(expense.id === action.id){
                    return{
                        ...expense,
                        ...action.updates
                    }
                }else{
                    return expense;
                }
            })
        default:
            return state;
    }
}

export {expensesReducer as default}