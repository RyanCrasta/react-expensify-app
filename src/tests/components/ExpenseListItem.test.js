import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem.js'
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem',() => {
    const wrapper = shallow(<ExpenseListItem expenses={[expenses[0]]}/>);
    expect(wrapper).toMatchSnapshot();
})


