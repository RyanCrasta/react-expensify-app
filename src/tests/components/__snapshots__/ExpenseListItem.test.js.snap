// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <div
    key="1"
  >
    <Link
      to="/edit1"
    >
      <h3>
        juice
      </h3>
    </Link>
    45
     - 
    0
    <p />
  </div>
</div>
`;
